src/components/SourcesTree.js

```diff
@@ -54,6 +55,15 @@ let SourcesTree = React.createClass({
       return this.setState({ listItems });
     }
 
+    if (nextProps.selectedSource != this.props.selectedSource) {
+      const hLightItems = getDirectories(
+        nextProps.selectedSource.get("url"),
+        this.state.sourceTree
+      );
+
+      return this.setState({ hLightItems });
```

Comment:
hmm, it would be nice be able to figure-out here or in managed tree which is the most specific visible node and just highlight that.

it's weird to have many highlight items